tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
```

Comment:
1. I think the comments are redundant, we can always hover the property in VSCode and it will tell us its description.
2. Let's make the comipler more strict by:
    1. Removing `allowJS`
    1. Adding `"strict": true`

Reply:
@a7med-mahmoud Enabling the `strict` flag is equivalent to turning on all of the _strict mode family options_, which are mostly included in your comment. There's no need to add them individually (unless we want to disable certain ones).

Reply:
@DavidMina96 Looking at the docs it seems to enable the following flags:

- `alwaysStrict`
- `strictBindCallApply`
- `strictFunctionTypes`
- `strictNullChecks`
- `strictPropertyInitialization`
- `useUnknownInCatchVariables`
- `noImplicitAny`
- `noImplicitThis`

This leaves us with:

```json
"noUnusedLocals": true,
"noUnusedParameters": true,
"noImplicitReturns": true,
"noFallthroughCasesInSwitch": true
```

What do you think?
We don't have to add them all but preferably the first two as they can be very useful.

Reply:
@a7med-mahmoud They can be useful, but I don't see an urgency to add them in this PR as it's somehow not within its core scope (btw there are no errors in the current code that would be captured with these two flags). Maybe we can consider the strictness in another PR later should we need that.